cordova-fetch/index.js

```diff
@@ -148,9 +148,15 @@ function trimID(target) {
         target = parts[1];
     }
 
-    //If local path exists, set target to final directory
+    // If local path exists, try to get plugin id from package.json or set target to final directory
     if (fs.existsSync(target)) {
-        target = path.basename(target);
+        var pluginId, pkgJsonPath = path.join(target, 'package.json');
+
+        if (fs.existsSync(pkgJsonPath)) {
+            pluginId = JSON.parse(fs.readFileSync(pkgJsonPath)).cordova.id;
```

Comment:
I think it was due to packageJson.name potentially being different then plugin.xml id field. But I don't know of any plugins with this and I'm not sure it makes sense to have that distinction anymore.